cmd/provenanced/cmd/init.go

```diff
@@ -139,53 +123,142 @@ func Init(
 	if err != nil {
 		return err
 	}
-	genFile := config.GenesisFile()
-	if !viper.GetBool(FlagOverwrite) && tmos.FileExists(genFile) {
-		return fmt.Errorf("genesis.json file already exists: %v", genFile)
-	}
-	if err := initGenFile(cdc, mbm, config.Moniker, chainID, nodeID, genFile); err != nil {
+
+	mustFprintf(cmd.OutOrStdout(), "node id: %s\n", nodeID)
+
+	if err = initGenFile(cmd, cdc, mbm, isTestnet, chainID, genFile); err != nil {
 		return err
 	}
-	tmconfig.WriteConfigFile(filepath.Join(config.RootDir, "config", "config.toml"), config)
+
+	configFile := filepath.Join(config.RootDir, "config", "config.toml")
+	tmconfig.WriteConfigFile(configFile, config)
+	mustFprintf(cmd.OutOrStdout(), "Tendermint config file updated: %s\n", configFile)
+
+	clientConfigFile := filepath.Join(config.RootDir, "config", "client.toml")
+	clientConfig, crerr := clientconf.GetClientConfig(clientConfigFile, client.GetClientContextFromCmd(cmd).Viper)
+	if crerr != nil {
+		return fmt.Errorf("couldn't get client config: %v", crerr)
+	}
+	clientConfig.ChainID = chainID
+	clientConfig.Node = config.RPC.ListenAddress
+	clientconf.WriteConfigToFile(clientConfigFile, clientConfig)
+	mustFprintf(cmd.OutOrStdout(), "Client config file updated: %s\n", clientConfigFile)
+
 	return nil
 }
 
-func initGenFile(cdc codec.JSONCodec, mbm module.BasicManager, moniker, chainID, nodeID string, genFile string) error {
+func initGenFile(
+	cmd *cobra.Command,
+	cdc codec.JSONCodec,
+	mbm module.BasicManager,
+	isTestnet bool,
+	chainID, genFile string,
+) error {
+	minDeposit := int64(1000000000000)  // 1,000,000,000,000
+	downtimeJailDurationStr := "86400s" // 1 day
+	maxValidators := uint32(20)
+	maxGas := int64(60000000) // 60,000,000
+	unrestrictedDenomRegex := `[a-zA-Z][a-zA-Z0-9\-\.]{7,64}`
+	if isTestnet {
+		mustFprintf(cmd.OutOrStdout(), "Using testnet defaults\n")
+		minDeposit = 10000000            // 10,000,000
+		downtimeJailDurationStr = "600s" // 10 minutes
+		maxValidators = 100
+		maxGas = -1
+		unrestrictedDenomRegex = `[a-zA-Z][a-zA-Z0-9\-\.]{2,64}`
```

Comment:
This was changed on testnet as well as mainnet to implement the restricted marker denom system.  Both require 8 char minimum now.  A question is also if the mixed case denom should be allowed vs lower case only...